Makefile

```diff
@@ -191,7 +191,17 @@ GOLANGCI_LINT_BIN := golangci-lint
 GOLANGCI_LINT := $(abspath $(TOOLS_BIN_DIR)/$(GOLANGCI_LINT_BIN))
 
 # Define Docker related variables. Releases should modify and double check these vars.
-REGISTRY ?= gcr.io/$(shell gcloud config get-value project)
+ifneq (,$(shell command -v gcloud))
+	ifneq (,$(shell gcloud config get-value project))
```

Comment:
Should we assign this command to a temp variable, like @nawazkh did in kubernetes-sigs/cluster-api-provider-azure#5221?